app/assets/javascripts/tylium/pages/activities.js

```diff
@@ -0,0 +1,62 @@
+document.addEventListener('turbolinks:load', function() {
+  var $activitiesGroupsContainer = $('[data-behavior="activities-groups-container"]');
+  var loading = false;
+  var canLoadMore = true;
+
+  if ($activitiesGroupsContainer.length) {
+    var $viewContent = $('[data-behavior="view-content"]');
+
+    $viewContent.on('scroll', function() {
+      var viewHeight = $viewContent.height();
+      var scrollTop = $viewContent.scrollTop();
+      var scrollHeight = $viewContent[0].scrollHeight;
+
+      // 64px is the sum of margin bottom and padding bottom of the content-container class
+      if (canLoadMore && !loading && viewHeight + scrollTop >= (scrollHeight - 64)) {
```

Comment:
Can we make this functionality a generic class? The functionality seems useful and maybe i'm getting ahead of myself (@MattBudz ?) but if there's anywhere else we could use this (notifications?) then making a generic Infinite Scroll module would be useful.

If it's something we would like I think most of this could stay the same. Once an infinite scroll container is scoped stuff like `activities-spinner` would just be `spinner`. The module would be initialized with `[data-behavior="infinite-scroll"]` and the container would have properties for `url` etc.

Reply:
@brianp I think we can extract it once we find another use for it. I don't think we need to over-engineer this just in case we may need it at some point elsewhere else.

I do like this idea, I just don't think it's necessary right now.

Reply:
@brianp I think this can be easily extracted out, but i agree with @MattBudz  that we should do that when it is required in multiple pages.